Makefile.core.mk

```diff
@@ -344,6 +344,17 @@ shellcheck:
 MARKDOWN_LINT_WHITELIST=localhost:8080,storage.googleapis.com/istio-artifacts/pilot/,http://ratings.default.svc.cluster.local:9080/ratings
 
 lint_modern: lint-python lint-copyright-banner lint-scripts lint-dockerfiles lint-markdown lint-yaml
+	@GOGC=20 golangci-lint run -c ./common/config/.golangci.yml ./galley/...
```

Comment:
did you mean to keep this around? shouldn't we be doing the `lint-go`? I suspect this will be slower and we miss a lot of folders

Reply:
Yes, this intentional. The lint-go will not work for most users due to the enormous memory consumption. This approach consumes 1/6th as much memory.

Reply:
We had lint in this repo for a while without info, why do we need this now?

@sdake did a lot of work on this

Reply:
@howardjohn Because this will now be running in a container for all users, and as such it would fail in most people's docker setup. And even without a container, this has been using an insane amount of memory. With GOGC=1, memory consumption peeks at 16G, so you need that much free at one time to run the linters. With this change, it peeks at 5G which is still a lot, but much more practical.

I took out the GOGC=20 which substantially reduces the amount of time spent in the garbage collector overall. And the reduced memory footprint also means substantially faster gcs when they do happen. I haven't measured the end-to-end time explicitly, but it feels as though it's in the same ballpark.

In terms of missing directories, it's a possibility. But we'll be moving stuff out of istio/istio over time and at some point we'll be able to lint the repo as a whole again. In the meantime, we have to watch for the introduction of new top-level directories (which is really a quite infrequent thing)